src/assets/AssetsDetectionController.ts

```diff
@@ -104,6 +144,25 @@ export class AssetsDetectionController extends BaseController<AssetsDetectionCon
       tokens: [],
     };
     this.initialize();
+    this.getAssetsState = getAssetsState;
+    this.addTokens = addTokens;
+    onAssetsStateChange(({ tokens }) => {
+      this.configure({ tokens });
+    });
+    onPreferencesStateChange(({ selectedAddress }) => {
+      const actualSelectedAddress = this.config.selectedAddress;
```

Comment:
just a question here, config is going to be deprecated in next updates?

Reply:
It will be removed in the new base controller. But we'll be migrating each controller to that one at a time. Any controllers still extending the old base controller will continue to use config in the interim.

Though, in this particular case, I'm not sure we need to use the config to store `selectedAddress`  :thinking:

Reply:
I'm going to leave this as-is for now, as keeping this in the config does still serve a purpose. It makes it easier to see when the address has changed, v.s. other state. This is easier to detect with the new base controller API, but with the current one this is a reasonable solution.